src/phoenix/datasets/evaluators.py

```diff
@@ -61,6 +67,150 @@ def evaluate(self, example: Example, exp_run: ExperimentRun) -> EvaluationResult
             ),
         )
 
+    async def async_evaluate(self, example: Example, exp_run: ExperimentRun) -> EvaluationResult:
+        return self.evaluate(example, exp_run)
+
+
+class LLMCriteriaEvaluator(ExperimentEvaluator):
+    annotator_kind = "LLM"
+    _base_template = (
+        "Determine if the following text is {criteria}. {description}"
+        "First, explain step-by-step why you think the text is or is not {criteria}. Then provide "
+        "a single word label; 'true' if the text is {criteria} or 'false' if the text is not "
+        "{criteria}. Here is an example template for whether the text meets a criteria:\n\n"
+        "CRITERIA: the text is '{criteria}'\n"
+        "TEXT: *the provided text to evaluate*\n"
+        "EXPLANATION: *a step by step explanation of your reasoning for whether the text meets "
+        "the criteria*\n"
+        "LABEL: *true or false*\n\n"
+        "Follow this template for the following text:\n\n"
+        "CRITERIA: the text is '{criteria}'\n"
+        "TEXT: {text}\n"
+        "EXPLANATION: "
+    )
+    _description = "In this context, '{criteria}' means the text '{description}'. "
+
+    def __init__(
+        self,
+        model: LLMBaseModel,
+        criteria: str,
+        description: Optional[str],
+        name: Optional[str] = None,
+    ):
+        self.model = model
+        self.criteria = criteria
+        self.description = description
+        self.template = self._format_base_template(self.criteria, self.description)
+        if name is not None:
+            self.name = name
+        else:
+            name = "LLMCriteriaEvaluator"
```

Comment:
I think we should make the name required, otherwise it won't make sense in the UI.